### DISCUSSION TITLE
Where does dnscrypt-proxy store cashed dns and how to clear it (windows dnscache service is disabled) ?

### DISCUSSION THREAD

**Discussion by maestropss** [DNSCrypt/dnscrypt-proxy#2183] [Q&A]
Hello. Where does dnscrypt-proxy store cashed dns and how to clear it (windows dnscache service is disabled)?
thanks

**Comment by jedisct1** [comment]
In memory!

So, to clear it, just restart the service.

**Reply by maestropss** [reply]
Hem! This means that IF an internet connection is often reestablished on my laptop as I turn off my laptop often during a day, THAN the cache function is literally useless for longer TTL (as suggested by Frank in this article https://00f.net/2019/11/03/stop-using-low-dns-ttls/). Am I right? 

Maybe it is reasonable to use permanent storage, like SSD, to keep the DNS cache?

**Reply by jedisct1** [reply]
If you put the laptop into sleep mode, the cache is not lost.

But with most TTLs still being very low these days, a persistent cache would not be really helpful.

**Reply by syphyr** [reply]
Writing the cache to disk would also not be very good on some devices (such as android) because it would cause a lot of wear on the device's limited storage.

**Reply by maestropss** [reply]
Thank you for your responses. Do I get it right : if I put the laptop in the hibernate mode, the cache will not be lost either (the same as it is not in the sleep mode)?

**Reply by jedisct1** [reply]
It will not!
